compiler-tests/src/test/java/com/squareup/anvil/compiler/dagger/InjectConstructorFactoryGeneratorTest.kt

```diff
@@ -0,0 +1,2819 @@
+package com.squareup.anvil.compiler.dagger
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.compiler.testing.CompilationModeTest
+import com.squareup.anvil.compiler.testing.reflect.createInstance
+import com.squareup.anvil.compiler.testing.reflect.factoryClass
+import com.squareup.anvil.compiler.testing.reflect.getFieldValue
+import com.squareup.anvil.compiler.testing.reflect.injectClass_Factory
+import com.squareup.anvil.compiler.testing.reflect.isStatic
+import com.squareup.anvil.compiler.testing.skipWhen
+import dagger.Lazy
+import dagger.internal.Factory
+import io.kotest.matchers.shouldBe
+import org.jetbrains.kotlin.cli.common.ExitCode
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.TestFactory
+import java.io.File
+import javax.inject.Provider
+
+class InjectConstructorFactoryGeneratorTest : CompilationModeTest() {
```

Comment:
It looks like this is duplicated instead of moved right now, can we delete the original? This will help to prevent unintentional diverging between the two as well as preserve the git history which can be really helpful for some test cases. Also with this test being as large as it is, it will help to highlight and confirm what the new changes are.

Reply:
I hadn't noticed this in the diff.  I'd certainly prefer to keep its git history.  I'll figure out what I need to do to fix that.

Reply:
I split it into two different commits.  It still doesn't present nicely in the PR diff, but the history is preserved. 🤷 

<img width="511" alt="image" src="https://github.com/user-attachments/assets/50ff9082-b5b8-46ac-9747-5f10bd98ffa3" />

Reply:
Awesome thanks for taking the time to adjust it! Unfortunately yea sometimes github doesn't handle the diff the same as git goes